napari/_qt/layer_controls/qt_tracks_controls.py

```diff
@@ -99,7 +100,6 @@ def __init__(self, layer):
 
         self._on_tail_length_change()
         self._on_tail_width_change()
-        self._on_properties_change()
```

Comment:
@quantumjot , I removed this call to the properties change callback. The `color_by` dropdown is populated earlier in the `__init__` method, so it seems safe to remove. It was causing the `color_by` value set in the layer init to be overwritten due to line 135 below. When the combox is cleared, the `textChanged` callback is triggered, which tries to update `layer.color_by`.

https://github.com/napari/napari/blob/a8f8eaa928445d5080683188ae9943f16d39f481/napari/_qt/layer_controls/qt_tracks_controls.py#L132-L136

Reply:
I think it makes a lot of sense to remove it! The properties change needs to reset the `color_by` combobox since the user *could* change the properties dict once the layer is instantiated.

Reply:
Great! Thanks for the quick feedback, @quantumjot ! I totally agree we need to keep the `_on_properties_change()` method for when the properties are altered on the model after initialization.

On a related note, a comment about the `_on_properties_change()` method: I think we may want to block the`textChanged` event from being emitted following the `color_by_combobox.clear()`. I don't think it's causing any issues now, but I think it's generally good to avoid unnecessary events/callbacks. Both for performance reasons and I find following the code through events/callbacks when debugging challenging (thus minimizing them reduces places to check). I need to check the qt docs, but I am pretty sure it's possible to temporarily block specific signals.

Reply:
That makes a lot of sense. Glad I'm not the only one that finds the events and callbacks challenging to debug!